etl/databases/embedding_database.py

```diff
@@ -0,0 +1,74 @@
+from pathlib import Path
+from typing import Self
+
+from langchain.docstore.document import Document
+from langchain_community.vectorstores import VectorStore
+
+from etl.databases import Database
+from etl.pipelines import OpenaiEmbeddingPipeline
+from etl.resources import OpenaiSettings
+
+
+class EmbeddingDatabase(Database):
+    """A database that holds an embedding store."""
+
+    def __init__(
+        self,
+        *,
+        embedding_store: VectorStore,
+        embeddings_cache_directory_path: Path,
+    ) -> None:
+        self.__embedding_store = embedding_store
+        self.__embeddings_cache_directory_path = embeddings_cache_directory_path
+
+    @classmethod
+    def create(
+        cls,
+        *,
+        documents: tuple[Document, ...],
+        openai_settings: OpenaiSettings,
+        embeddings_cache_directory_path: Path,
+    ) -> Self:
+        """
+        Return an EmbeddingDatabase that contains an embedding store created by an OpenaiEmbeddingPipeline.
+        """
+        return cls(
+            embedding_store=OpenaiEmbeddingPipeline(
+                openai_settings=openai_settings,
+                openai_embeddings_cache_directory_path=embeddings_cache_directory_path,
+            ).create_embedding_store(documents=documents),
+            embeddings_cache_directory_path=embeddings_cache_directory_path,
+        )
+
+    @property
+    def descriptor(self) -> Database.Descriptor:
+        """The handle of the embedding store in the EmbeddingDatabase."""
+
+        return EmbeddingDatabase.Descriptor(self.__embeddings_cache_directory_path)
+
+    @classmethod
+    def open(
```

Comment:
The main purpose of the create/open logic is to avoid populating the store multiple times.

It goes something like this:

```
@asset
def store() -> Store.Descriptor:
    with Store.create() as store:
        store.load(from_file_path)
        return store.descriptor

@asset
def downstream_asset(store: Store.Descriptor):
    with Store.open(store) as open_store:
        ... Use open store, it's already been loaded ...
```

The descriptor points at the persisted, loaded store. With Oxigraph that'd be an Oxigraph directory path. With Postgres it'd be a connection string to a previously-loaded database with an ORM on top. Etc.